src/wolfio.c

```diff
@@ -153,6 +153,11 @@ static WC_INLINE int TranslateReturnCode(int old, int sd)
         if (errno == RTCSERR_TCP_TIMED_OUT)
             errno = SOCKET_EAGAIN;
     }
+#elif defined(WOLFSSL_EMNET)
+    if (old < 0) { /* SOCKET_ERROR */
+        /* Get the real socket error */
+        getsockopt(sd, SOL_SOCKET, SO_ERROR, &old, (int)sizeof(old));
```

Comment:
This actually calls: `IP_SOCK_getsockopt` and API is `int IP_SOCK_getsockopt(int hSock, int Level, int Name, void* pVal, int ValLen)`. Since the `getsockopt` is not fully compatible I might change this to use `IP_SOCK_getsockopt` directly. That API is included via `IP/IP.h`.